Outreachyp

damli40 opened the pull request:
My first ever Pull request, so excited..yay :)....hope it gets merged

betatim commented:
Hi 👋 

thanks for opening a pull request, your first one!

To help review PRs and understand their context it is helpful for PR authors to include a short description of why they made the changes they made and if there are a lot of changes a short summary of them. Generally a good assumption to make is that the people reviewing your PR will know nothing about it (and you) except what is written in the PR itself. SO it is helpful to include some information that helps them make a decision or give advice on what to do next with the PR to move it closer to being merged.

Please also change the title of the PR to something more descriptive.

damli40 commented:
> Hi 👋 
> 
> 
> 
> thanks for opening a pull request, your first one!
> 
> 
> 
> To help review PRs and understand their context it is helpful for PR authors to include a short description of why they made the changes they made and if there are a lot of changes a short summary of them. Generally a good assumption to make is that the people reviewing your PR will know nothing about it (and you) except what is written in the PR itself. SO it is helpful to include some information that helps them make a decision or give advice on what to do next with the PR to move it closer to being merged.
> 
> 
> 
> Please also change the title of the PR to something more descriptive.

Thank you so much, I have made the necessary changes!

consideRatio commented:
Thanks @damli40 for your work on this!! :heart: :tada: :sunflower:

There are quite a bit of new things for me to learn related to this!

- I understood the `alt=` part, :+1: 
- I don't yet understand what the `.hintrc` file does or what tools would be influenced by it. Is it for your use with WAVE? Note that I have never used WAVE and have no knowledge about it.
  If the `.hintrc` file is to be added, I'd like this PR to include a clarification of what it does and why its added. Ideally the `.hintrc` file itself should have a comment to clarify its purpose if it supports comments. For example [like the `.flake8` file has](https://github.com/jupyterhub/binderhub/blob/master/.flake8)
- I tried to learn what `role="list"` was about and found notes about ARIA roles. It seems from [this documentation on the `<ul>` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul) that there is an implicit ARIA `role="list"` already, so maybe we don't need to specify that? I'm not sure, what do you think?

consideRatio commented on binderhub/templates/page.html:
```diff
@@ -1,6 +1,8 @@
 <!DOCTYPE html>
-<html>
+<html lang="en"> 
+  <!-- adding language description for accessibility  -->
```
This comment can be part of the pull request description or a commit message adding it, but I think its better omitted from the HTML code itself.

Below is a code suggestion to remove it using the GitHub user interace!

```suggestion
```

damli40 commented:
> Thanks @damli40 for your work on this!! :heart: :tada: :sunflower:
> 
> 
> 
> There are quite a bit of new things for me to learn related to this!
> 
> 
> 
> - I understood the `alt=` part, :+1: 
> 
> - I don't yet understand what the `.hintrc` file does or what tools would be influenced by it. Is it for your use with WAVE? Note that I have never used WAVE and have no knowledge about it.
> 
>   If the `.hintrc` file is to be added, I'd like this PR to include a clarification of what it does and why its added. Ideally the `.hintrc` file itself should have a comment to clarify its purpose if it supports comments. For example [like the `.flake8` file has](https://github.com/jupyterhub/binderhub/blob/master/.flake8)
> 
> - I tried to learn what `role="list"` was about and found notes about ARIA roles. It seems from [this documentation on the `<ul>` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul) that there is an implicit ARIA `role="list"` already, so maybe we don't need to specify that? I'm not sure, what do you think?

Thank you so much for the response!
- I've commented what the '.hintrc' is for but it's basically a configuration file that checks the whole HTML code and suggests accessibility changes. You can find out more here https://webhint.io/docs/user-guide/

- Yes! You're correct, the "role=list" is not needed. I must have misread the rules on how to use but I understand now thanks to you.

- Lastly, I've removed the comment and added it to the commit message instead!


All in all, thank you so much, learning and giving knowledge is why I fell in love with open source! 

Hopefully, with these corrections the PR can get merged!❤️

minrk commented:
Thank you for the pull request!

In general in the future, it's a good idea to separate PRs that add tool configurations from changes to existing files. Usually, if a tool configuration is not present in a project, discussion about adopting the tool in the project should happen first, because as soon as a configuration file is added, it's up to the project maintainers to _maintain_ that configuration file, along with all the others. If we don't actually use the tool, it can easily become stale and its presence misleading.

I've resolved conflicts with another PR that made some of the same changes.

manics commented:
https://webhint.io/docs/user-guide/#customize-webhint-in-your-project
> A custom hint (.hintrc file) is useful when you want either of the following actions.
>
>    Ignore specific errors.
>    Highlight or break for specific warnings.

Does this mean we should automatically run it in CI with a suitable config?

damli40 commented:
> Thank you for the pull request!
> 
> 
> 
> In general in the future, it's a good idea to separate PRs that add tool configurations from changes to existing files. Usually, if a tool configuration is not present in a project, discussion about adopting the tool in the project should happen first, because as soon as a configuration file is added, it's up to the project maintainers to _maintain_ that configuration file, along with all the others. If we don't actually use the tool, it can easily become stale and its presence misleading.
> 
> 
> 
> I've resolved conflicts with another PR that made some of the same changes.

Thank you @minrk !!! Finally. This has been a great learning process, all corrections have been noted. Thanks again

minrk commented:
@manics 

> Does this mean we should automatically run it in CI with a suitable config?

I think we will want to get there, yes. Exactly what config to use and how to integrate it in CI is a goal of the outreachy project. It may, for example, require jinja template rendering first.
